### PR TITLE
add pip check to recipe tests

### PR DESCRIPTION
**Pull request by dholth**
### Description

lief was added to the pip-style dependencies, causing pip check to fail since py-lief's conda package has no `dist-info` (pip metadata)

Xref https://github.com/conda/conda-build/issues/5101

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -78,6 +78,7 @@ test:
     - conda debug --help
     # bdist_conda
     - python test_bdist_conda_setup.py bdist_conda --help
+    - python -m pip check
```
Comment (kenodegard) [inline recipe/meta.yaml:81]: Move this to the top of the commands list (otherwise the comments don't make sense)

Comment (kenodegard) [reply recipe/meta.yaml:81]: Also need to add pip to `test/requires`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -51,6 +51,8 @@ requirements:
     - six
     - tomli                        # [py<311]
     - tqdm
+  test:
+    - pip
```
Comment (kenodegard) [inline recipe/meta.yaml:55]: this belongs in the `test` section below

**Review by kenodegard — DISMISSED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -36,7 +36,6 @@ dependencies = [
   "jinja2",
   "jsonschema >=4.19",
   "libarchive-c",
-  "lief",
```
Comment (kenodegard) [inline pyproject.toml:39]:
```
# "lief",  # see https://github.com/conda/conda-build/issues/5101
```

Comment (dholth) [reply pyproject.toml:39]: I expect the canary section of this CI build to run the pip check test.

Comment (kenodegard) [reply pyproject.toml:39]: canaries are only built for PRs if the https://github.com/conda/conda-build/labels/build%3A%3Areview label is applied

**Review by kenodegard — APPROVED** [review]

File: recipe/meta.yaml [hunk 4]
```diff
@@ -61,9 +61,13 @@ test:
     - conda_build
     # new/updated submodules (can be dropped after 1-2 releases)
     - conda_build.index
+  requires:
+    - setuptools
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:65]: This feels weird to me. Is it because pip check complains that setuptools isn't present?

Comment (dholth) [reply recipe/meta.yaml:65]: setuptools was in the other version of the recipe/feedstock. Maybe for `- test_bdist_conda_setup.py`

Comment (kenodegard) [reply recipe/meta.yaml:65]: this is copied over from the AnacondaRecipes conda-build-feedstock where it was originally added in https://github.com/AnacondaRecipes/conda-build-feedstock/pull/26, not sure why

**Pull request merged**